client/components/mma/cancel/cancellationSaves/supporterplus/SupporterPlusOffer.tsx

```diff
@@ -0,0 +1,302 @@
+import { css } from '@emotion/react';
+import {
+	from,
+	neutral,
+	palette,
+	space,
+	textEgyptian17,
+	textSans12,
+	textSans15,
+	textSans17,
+	textSans20,
+	textSansBold28,
+} from '@guardian/source/foundations';
+import { Button } from '@guardian/source/react-components';
+import { capitalize } from 'lodash';
+import { useContext } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { Ribbon } from '@/client/components/shared/Ribbon';
+import { measure } from '@/client/styles/typography';
+import type { DiscountPreviewResponse } from '@/client/utilities/discountPreview';
+import { parseDate } from '@/shared/dates';
+import { number2words } from '@/shared/numberUtils';
+import { getMainPlan, isPaidSubscriptionPlan } from '@/shared/productResponse';
+import type { DeliveryRecordDetail } from '../../../delivery/records/deliveryRecordsApi';
+import type { OutstandingHolidayStop } from '../../../holiday/HolidayStopApi';
+import { BenefitsSection } from '../../../shared/benefits/BenefitsSection';
+import { Heading } from '../../../shared/Heading';
+import { ProgressStepper } from '../../../shared/ProgressStepper';
+import type { CancellationContextInterface } from '../../CancellationContainer';
+import { CancellationContext } from '../../CancellationContainer';
+import type { OptionalCancellationReasonId } from '../../cancellationReason';
+
+interface RouterSate extends DiscountPreviewResponse {
+	selectedReasonId: OptionalCancellationReasonId;
+	cancellationPolicy: string;
+	caseId: string;
+	holidayStops?: OutstandingHolidayStop[];
+	deliveryCredits?: DeliveryRecordDetail[];
+	eligibleForFreePeriodOffer: boolean;
+}
+
+export const SupporterPlusOffer = () => {
+	const location = useLocation();
+	const routerState = location.state as RouterSate;
+	const navigate = useNavigate();
+
+	const cancellationContext = useContext(
+		CancellationContext,
+	) as CancellationContextInterface;
+
+	const productDetail = cancellationContext.productDetail;
+	const mainPlan = getMainPlan(productDetail.subscription);
+
+	const offerPeriodWord = number2words(routerState.upToPeriods);
+	const offerPeriodType = routerState.upToPeriodsType.toLowerCase();
+	const nextNonDiscountedPaymentDate = parseDate(
+		routerState.nextNonDiscountedPaymentDate,
+		'yyyy-MM-dd',
+	).dateStr();
+
+	const standfirstCss = css`
```

Comment:
Do these emotion styles need to be defined in the function body of the component? I'm sure performance wise it's no problem, but would it make the component clearer if these were pulled outside (as they don't need to be defined on every render)? Obvs feel free to ignore, especially if this is the convention in this repo!

Reply:
yeah, that's a good shout re re render. I'll move them out :+1: